src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -2898,9 +2913,9 @@ window.qBittorrent.DynamicTable ??= (() => {
                     const node1 = this.getNode(row1.data.rowId);
                     const node2 = this.getNode(row2.data.rowId);
                     if (node1.isFolder && !node2.isFolder)
-                        return -1;
-                    if (node2.isFolder && !node1.isFolder)
                         return 1;
+                    if (node2.isFolder && !node1.isFolder)
+                        return -1;
```

Comment:
How come we need to invert this completely now? Wasn't it working correctly before #22502?

Reply:
>Now it's always folder first, which also matches GUI behaviour.

The folder first behavior is correct but the change here is unusual. The convention is to return `-1` if `left` should be placed before `right`  (left = node1, right = node2). Your change inverted it.  I suspect you should fix other places instead.

Reply:
Sounds like `_filterNodes` should be revised. First, away from recursion and keep the list non-reversed.
I can take a look at it but it will wait until the weekend. In the meantime, feel free to fix it too.

Reply:
Try changing this line 
https://github.com/qbittorrent/qBittorrent/blob/0b3bce89933c0323b1cd4309ace310af4c309a21/src/webui/www/private/scripts/dynamicTable.js#L2918
to `const childAdded = node.children.toReversed().reduce((acc, child) => {`
and revert the changes to `_sortNodesByColumn` function.
This should fix the sorting problem, please verify.

I'll submit non-recursive in another PR later.